### DISCUSSION TITLE
Ubuntu suffixed Dated Codename versions

### DISCUSSION THREAD

**Discussion by wparr-circle** [renovatebot/renovate#30560] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Recently we noticed that renovate didn't pick up the latest version of Jammy from upstream dockerhub - see: https://hub.docker.com/layers/library/ubuntu/jammy-20240627.1/images/sha256-0eb0f877e1c869a300c442c41120e778db7161419244ee5cbc6fa5f134e74736?context=explore

We use custom versioning for these, as opposed to using renovate [ubuntu versioning](https://docs.renovatebot.com/modules/versioning/ubuntu/) - and had thought we might have missed an edge case that would be resolved by using the ubuntu versioning.

Looking into the implementation... It looks like we renovate will never pick up an ubuntu version tagged this way either.

ie. https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/ubuntu/common.ts#L3

The following regex is used, which would never match on `jammy-20240627.1` due to the suffixed `.1`
```
/^(?<codename>\w+)-(?<date>\d{8})$/
```

We haven't been able to find details of what exactly this means on the dated codename version so far - but it doesn't look to be a mistake on the part of the maintainers _and_ there are historical releases matching this as well.

Should renovate support matching this also?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Sure, PR welcome

**Reply by HonkingGoose** [reply, marked as answer]
See issue:

- #30667

@wparr-circle if you want to fix this problem, please comment on the _issue_, so I can assign it to you. :wink:
